docs/Displaying a User Interface/Template Buttons/index.md

```diff
@@ -0,0 +1,561 @@
+	public void onComplete(boolean success) {
+		Log.i(TAG, "ScreenManager update complete: " + success);
+	}
+});
+
+// Transition to a new state
+SoftButtonObject retrievedSoftButtonObject = sdlManager.getScreenManager().getSoftButtonObjectByName("softButtonObject");
+retrievedSoftButtonObject.transitionToNextState();
+```
+!@
+
+### Deleting Soft Buttons
+To delete soft buttons, simply pass the screen manager a new array of soft buttons. To delete all soft buttons, simply pass the screen manager an empty array.
+
+@![iOS]
+##### Objective-C
+```objc
+self.sdlManager.screenManager.softButtonObjects = @[];
+```
+
+##### Swift
+```swift
+sdlManager.screenManager.softButtonObjects = []
+```
+!@
+
+@![android,javaSE,javaEE]
+```java
+sdlManager.getScreenManager().setSoftButtonObjects(Collections.EMPTY_LIST);
+```
+!@
+
+## Subscribing to System Buttons
+Subscribe buttons are used to detect when your user has interacted with buttons located in the car's center console or steering wheel. You can subscribe to the following buttons:
+
+| Button  | Template |
+| ------------- | ------------- |
+| Play / Pause | Media only |
+| Ok | Media only |
+| Seek left | Media only |
+| Seek right | Media only |
+| Tune up | Media only |
+| Tune down | Media only |
+| Preset 0-9 | Any |
+| Search | Any |
+| Center Location | NavigationFullscreenMap only |
+| Zoom In | NavigationFullscreenMap only |
+| Zoom Out | NavigationFullscreenMap only |
+| Pan Up | NavigationFullscreenMap only |
+| Pan Up-Right | NavigationFullscreenMap only |
+| Pan Right | NavigationFullscreenMap only |
+| Pan Down-Right | NavigationFullscreenMap only |
+| Pan Down | NavigationFullscreenMap only |
+| Pan Down-Left | NavigationFullscreenMap only |
+| Pan Left | NavigationFullscreenMap only |
+| Pan Up-Left | NavigationFullscreenMap only |
+| Toggle Tilt | NavigationFullscreenMap only |
+| Rotate Clockwise | NavigationFullscreenMap only |
+| Rotate Counter-Clockwise | NavigationFullscreenMap only |
+| Toggle Heading | NavigationFullscreenMap only |
+
+!!! NOTE
+`Media` is the default template for media apps, and `NavigationFullscreenMap` is the default template for navigation apps.
+!!!
+
+### Subscribe Buttons HMI
+In the screenshot below, the pause, seek left and seek right icons are subscribe buttons.
+![Generic - Media Template with subscribe buttons](assets/Generic_template_media_light.png)
+
+!!! NOTE
+There is no way to customize a subscribe button's image or text.
+!!!
+
+### Audio-Related Buttons
+The play/pause, seek left, seek right, tune up, and tune down subscribe buttons can only be used in the `MEDIA` template. Depending on the manufacturer of the head unit, the subscribe button might also show up as a soft button in the media template. For example, the SYNC 3 HMI will add the ok, seek right, and seek left soft buttons to the media template when you subscribe to those buttons. You will automatically be assigned the media template if you set your app's `appType` to `MEDIA`.
```

Comment:
```suggestion
The play/pause, seek left, seek right, tune up, and tune down subscribe buttons can only be used in the `MEDIA` template. Depending on the head unit's OEM, the subscribed button could show up as a soft button in the media template, work as a hard button on the car console or steering wheel, or both. For example, the SYNC 3 HMI will add the play/pause, seek right, and seek left soft buttons to the media template when you subscribe to those buttons. However, it will only allow tune up and tune down to occur using the tune dial on the center console. You will automatically be assigned the media template if you set your app's `appType` to `MEDIA`.
```